x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/approximate/ConfidenceInterval.java

```diff
@@ -0,0 +1,342 @@
+            builder.appendNull();
+            return;
+        }
+        double bestEstimate = bestEstimateBlock.getInt(bestEstimateBlock.getFirstValueIndex(position));
+        double[] estimates = new double[estimatesBlock.getValueCount(position)];
+        for (int i = 0; i < estimatesBlock.getValueCount(position); i++) {
+            estimates[i] = estimatesBlock.getInt(estimatesBlock.getFirstValueIndex(position) + i);
+        }
+        int trialCount = trialCountBlock.getInt(trialCountBlock.getFirstValueIndex(position));
+        int bucketCount = bucketCountBlock.getInt(bucketCountBlock.getFirstValueIndex(position));
+        double confidenceLevel = confidenceLevelBlock.getDouble(confidenceLevelBlock.getFirstValueIndex(position));
+        double[] confidenceInterval = computeConfidenceInterval(bestEstimate, estimates, trialCount, bucketCount, confidenceLevel);
+        builder.beginPositionEntry();
+        for (double v : confidenceInterval) {
+            builder.appendInt((int) Math.round(v));
+        }
+        builder.endPositionEntry();
+    }
+
+    @Evaluator(extraName = "Long")
+    static void process(
+        LongBlock.Builder builder,
+        @Position int position,
+        LongBlock bestEstimateBlock,
+        LongBlock estimatesBlock,
+        IntBlock trialCountBlock,
+        IntBlock bucketCountBlock,
+        DoubleBlock confidenceLevelBlock
+    ) {
+        if (bestEstimateBlock.getValueCount(position) != 1
+            || trialCountBlock.getValueCount(position) != 1
+            || bucketCountBlock.getValueCount(position) != 1
+            || confidenceLevelBlock.getValueCount(position) != 1) {
+            builder.appendNull();
+            return;
+        }
+        double bestEstimate = bestEstimateBlock.getLong(bestEstimateBlock.getFirstValueIndex(position));
+        double[] estimates = new double[estimatesBlock.getValueCount(position)];
+        for (int i = 0; i < estimatesBlock.getValueCount(position); i++) {
+            estimates[i] = estimatesBlock.getLong(estimatesBlock.getFirstValueIndex(position) + i);
+        }
+        int trialCount = trialCountBlock.getInt(trialCountBlock.getFirstValueIndex(position));
+        int bucketCount = bucketCountBlock.getInt(bucketCountBlock.getFirstValueIndex(position));
+        double confidenceLevel = confidenceLevelBlock.getDouble(confidenceLevelBlock.getFirstValueIndex(position));
+        double[] confidenceInterval = computeConfidenceInterval(bestEstimate, estimates, trialCount, bucketCount, confidenceLevel);
+        builder.beginPositionEntry();
+        for (double v : confidenceInterval) {
+            builder.appendLong(Math.round(v));
+        }
+        builder.endPositionEntry();
+    }
+
+    public static double[] computeConfidenceInterval(
+        double bestEstimate,
+        double[] estimates,
+        int trialCount,
+        int bucketCount,
+        double confidenceLevel
+    ) {
```

Comment:
I think a comment explaining roughly what's happening here would be helpful, i.e.

1. `trialCount x "bucketCount random partitions of the original sample"` are generate
2. Distribution statistics are estimated for each sample and averaged
3. Parametric BCa is applied using mean and standard deviation to estimate the distribution function and its inverse
4. We correct for the sample mismatch, i.e. we assume both skew and standard deviation -> 0 as 1 / sqrt(n) for n samples used to estimate the statistic values